### PR TITLE
feat(worker, framework): Dynamic bridge url to not require local <> dev syncs

### PR DESCRIPTION
**Pull request by scopsy**
### What changed? Why was the change needed?

```
{
  "name": "hello-world-2",
  "to": {
    "subscriberId": "test-1",
    "email": "dima@dima.com",
    "phone": "+1"
  },
  "payload": {

  },
  "bridge": {
      "url": "https://aaaaaa.loca.lt/api/novu",
      "workflow": {
          "workflowId": "hello-world-2",
          "steps": [{
              "type": "sms",
              "stepId": "send-sms"
          }, {
              "type": "email",
              "stepId": "send-email"
          }]
      }
  }
}
```
### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: libs/application-generic/src/usecases/trigger-event/trigger-event.command.ts [hunk 1]
```diff
@@ -45,6 +45,11 @@ export class TriggerEventBaseCommand extends EnvironmentWithUserCommand {
   @IsOptional()
   @IsEnum(TriggerRequestCategoryEnum)
   requestCategory?: TriggerRequestCategoryEnum;
+
+  bridge?: {
```
Comment (SokratisVidros) [inline libs/application-generic/src/usecases/trigger-event/trigger-event.command.ts]: My suggestion is to keep them flat `bridge_url` and `workflowDefinition`

File: apps/api/src/app/events/usecases/parse-event-request/parse-event-request.command.ts [hunk 2]
```diff
@@ -45,11 +45,23 @@ export class ParseEventRequestMulticastCommand extends ParseEventRequestBaseComm
 
   @IsEnum(AddressingTypeEnum)
   addressingType: AddressingTypeEnum.MULTICAST;
+
+  @IsObject()
+  @IsOptional()
+  bridge?: {
+    url: string;
+  };
```
Comment (djabarovgeorge) [inline apps/api/src/app/events/usecases/parse-event-request/parse-event-request.command.ts]: bridgeUrl

File: apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts [hunk 3]
```diff
@@ -173,6 +182,12 @@ export class ParseEventRequest {
     };
   }
 
+  private isStatelessWorkflowAllowed(bridge: { url: string } | undefined): boolean {
+    const allowedEnvironment = process.env.NODE_ENV !== 'production';
```
Comment (djabarovgeorge) [inline apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts]: this should check the users environment, not process.env.NODE_ENV

File: apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts [hunk 4]
```diff
@@ -204,12 +204,13 @@ export class SendMessage {
...
   private async getStepExecutionHalt(
-    resonateSkip: boolean | undefined,
+    bridgeSkip: boolean | undefined,
     command: SendMessageCommand,
     variables: IFilterVariables
   ): Promise<{ filterResult: IConditionsFilterResponse | null; channelPreferenceResult: boolean | null }> {
-    if (resonateSkip === true) {
-      return { filterResult: null, channelPreferenceResult: null };
+    const skipHalt = this.shouldSkipHalt(bridgeSkip, command.job?.step?.bridgeUrl);
+    if (skipHalt) {
```
Comment (djabarovgeorge) [inline apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts:212]: Let's allow checking global user preferences here, and not skip them.

File: libs/application-generic/src/usecases/trigger-event/trigger-event.usecase.ts [hunk 5]
```diff
@@ -85,6 +114,10 @@ export class TriggerEvent {
         organizationId: mappedCommand.organizationId,
       });
 
+      const discoverWorkflow = await this.queryDiscoverWorkflow(command);
```
Comment (djabarovgeorge) [inline libs/application-generic/src/usecases/trigger-event/trigger-event.usecase.ts]: This should throw an error if not successfully, and also suggest to move this to the /events/trigger API call level so we can throw an HTTP error

File: libs/dal/src/repositories/notification/notification.schema.ts [hunk 6]
```diff
@@ -41,6 +41,9 @@ const notificationSchema = new Schema<NotificationDBModel>(
       type: Schema.Types.Mixed,
     },
     expireAt: Schema.Types.Date,
+    bridge: {
+      type: Schema.Types.Mixed,
```
Comment (djabarovgeorge) [inline libs/dal/src/repositories/notification/notification.schema.ts:45]: Should we be more structured here?

File: libs/application-generic/src/usecases/trigger-event/trigger-event.usecase.ts (hunk 5 above)
Comment (djabarovgeorge) [reply libs/application-generic/src/usecases/trigger-event/trigger-event.usecase.ts]: done

**Pull request merged**